text/0000-export-visibility.md

```diff
@@ -0,0 +1,837 @@
+- Feature Name: `export_visibility`
+- Start Date: 2025-06-12
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Documentation of
+[`#[no_mangle]`](https://doc.rust-lang.org/reference/abi.html#the-no_mangle-attribute)
+points out that by default a `#[no_mangle]` function (or `static`)
+will be "publicly exported from the produced library or object file".
+This RFC proposes a new `#[export_visibility = ...]` attribute
+to override this behavior.
+This means that if the same `#[no_mangle]` function is also
+decorated with `#[export_visibility = "target_default"]`,
+then it will instead use the default visibility of the target platform
+(which can be overriden with the
+[`-Zdefault-visibility=...`](https://doc.rust-lang.org/beta/unstable-book/compiler-flags/default-visibility.html)
+command-line flag).
```

Comment:
Do you mean that `-Zdefault-visibility=hidden` + `#[export_visibility = "target_default"]` on ELF would result in default visibility? If I understand that correctly, that makes this attribute useless for projects that let rustc do the linking and don't use `-Zdefault-visibility` as in that case there is still no way to restrict the visibility of `#[no_mangle]` functions.